### PR TITLE
Add integration tests for grafana deployment

### PR DESCRIPTION
**Pull request by ggriffiths**
**What this PR does / why we need it**:
Adds a few tests to the operator integration test suite for validating grafana deployments, svcs, configmaps, and airgapped support.

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ggriffiths** [comment]
Ran the test suite a few times, works as expected:
```
--- PASS: TestGrafana (173.22s)
    --- PASS: TestGrafana/GrafanaInstall (173.15s)
    --- PASS: TestGrafana/GrafanaInstallAirgapped (0.07s)
PASS
==================================================
--- PASS:
    GrafanaInstall [XXX XXX]
    GrafanaInstallAirgapped [XXX XXX]
--- SKIP:
--- FAIL:
==================================================

DONE 3 tests in 173.491s
Tests passed
```

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:2857]: Why are we validating this as a part of `validateStorkSecurityEnvVar()` function?

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:2937]: Can you add this as a part of `ValidateStorageCluster()` method

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:2962]: This doesn't actually validate Grafana Image, we should do it though as it is part of release manifest versions URL

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:3003]: Use this function as part of your `ValidateGrafanaDeployment()`, we need to validate images as a part of deployment validation, see example in the `ValidateStork()` or other components validation functions, also take a look at this function that does image validation inside containers https://github.com/libopenstorage/operator/blob/e4f90a46c0a2e28416d5acf44e953a5c42398010/pkg/util/test/util.go#L2577, this works with custom registries and trims image names accordingly using operator function `GetImageURN()`
NOTE: You need to validate images when they are part of the version image list https://install.portworx.com/3.0.0/version?kbver=1.24.0

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:2942]: This will skip validation even if your PX Operator version is 23.8, I thought you wanted to validate this starting with 23.8?

**Review by nikolaypopov — COMMENTED** [review]
Some changes are needed here, let me know if you want to sync up

**Comment by ggriffiths** [comment]
Tested again after addressing comments:
```
time="2023-07-25T19:53:35Z" level=info msg="Using version extracted from image name: 2.13.6"
time="2023-07-25T19:53:35Z" level=info msg="Update StorageCluster px-test-grafana in kube-system"
2023/07/25 19:53:35 grafana is not installed with image docker.io/grafana/grafana:8.5.27. number of pods: 0 Next retry in: 10s
time="2023-07-25T19:53:45Z" level=info msg="Grafana installed successfully with image docker.io/grafana/grafana:8.5.27"
--- PASS: TestGrafana (158.30s)
    --- PASS: TestGrafana/GrafanaInstall (143.24s)
    --- PASS: TestGrafana/GrafanaInstallVersionManifest (15.06s)
PASS
==================================================
--- PASS:
    GrafanaInstall [XXX XXX]
    GrafanaInstallVersionManifest [XXX XXX]
--- SKIP:
--- FAIL:
==================================================

DONE 3 tests in 158.637s
Tests passed
```

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:2972]: I would recommend to follow how we validate all expected images, from `pxImageList`, because that is what we want to be installed, because `DesiredImages` is what PX Operator decided to install, but it doesn't necessary match which what it should be installing. The correct images should come versions manifest, wether its `px-versions` configmap or URL endpoint

Comment (nikolaypopov) [reply pkg/util/test/util.go:2972]: Just pass the `pxImageList` to the function, find your Grafana image there, which should be `pxImageList["grafana"]`, this should give you the value of it like so `grafana: docker.io/grafana/grafana:7.5.17`, then use the `validateContainerImageInsidePods()` function, because it will correctly trim the image to the correct URN, incase of custom registry or anything like that

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:3052]: Remove this since you using `validateContainerImageInsidePods()` and we don't need this any more

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:2989]: Don't hardcode `kube-system`, this can be installed into any namespace, let's get namespace from `cluster.Namespace` and use that instead

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:2997]: Is grafana a deployment? Maybe just using `sched-ops` function `ValidateDeployment` should be enough here as it would do all the pod validation, if its not and its just a pod, this should be ok

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:3002]: Let's not log error and then return it, let's just return verbose error instead

File: pkg/util/test/util.go
Comment (nikolaypopov) [inline pkg/util/test/util.go:3013]: Same here and all other place you have this

**Review by nikolaypopov — COMMENTED** [review]
Overall looks much better, please address comments and build and image so I can run it in one of the basic jobs

File: pkg/util/test/util.go
Comment (ggriffiths) [reply pkg/util/test/util.go]: done

**Review by nikolaypopov — APPROVED** [review]
LGTM, let's run `op-basic` job once with this private image before merging. Let me know when you have an image ready

**Comment by ggriffiths** [comment]
Thanks, sent the test image.

**Comment by ggriffiths** [comment]
Running here:
https://jenkins.pwx.dev.purestorage.com/job/Operator/view/Operator%20Release%20Dashboard/job/op-basic-ocp/209

**Review by nikolaypopov — APPROVED** [review]

**Pull request merged**
